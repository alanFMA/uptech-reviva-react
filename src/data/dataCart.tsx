import { ICarrinhoProduto } from "../mocks/interface"

export const dataCart: ICarrinhoProduto[] = []