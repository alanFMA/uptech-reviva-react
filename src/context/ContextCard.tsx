import { createContext, ReactNode, useEffect, useState } from "react";
import { ICarrinhoProduto } from 'mocks/interface'
import { dataCart } from 'data/dataCart'

interface ContextDataCartProviderProps {
  dataCartValue: ICarrinhoProduto[];
  setDataCartValue: React.Dispatch<React.SetStateAction<ICarrinhoProduto[]>>;
}

interface ChildrenProps {
  children: ReactNode
}

export const ContextDataCart = createContext({} as ContextDataCartProviderProps);

export const ContextDataCartProvider = ({children}:ChildrenProps) => {
  const [dataCartValue, setDataCartValue] = useState(dataCart);

  return (
    <ContextDataCart.Provider value={{dataCartValue, setDataCartValue}}>
      {children}
    </ContextDataCart.Provider>
  );
};