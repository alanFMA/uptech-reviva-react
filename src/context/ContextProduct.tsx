import { createContext, ReactNode, useEffect, useState } from "react";
import { dataProduct } from "../data/dataProduct";
import { IProduto } from "mocks/interface"

interface ContextDataProductProviderProps {
  dataProductValue: IProduto[];
  setDataProductValue: React.Dispatch<React.SetStateAction<IProduto[]>>;
}

interface ChildrenProps {
  children: ReactNode
}

export const ContextDataProduct = createContext({} as ContextDataProductProviderProps);

export const ContextDataProductProvider = ({children}:ChildrenProps) => {
  const [dataProductValue, setDataProductValue] = useState(dataProduct);
  
  useEffect(() => {
    setDataProductValue(dataProduct);
  }, []);

  return (
    <ContextDataProduct.Provider value={{dataProductValue, setDataProductValue}}>
      {children}
    </ContextDataProduct.Provider>
  );
};