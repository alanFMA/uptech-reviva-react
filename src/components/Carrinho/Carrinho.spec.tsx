import { render } from "@testing-library/react";
import { ContextDataCart } from "context/ContextCard";
import { dataProduct } from "data/dataProduct";
import { useState } from "react";
import { Carrinho } from ".";

const Component = (props: any) => {
  const [dataCartValue, setDataCartValue] = useState(props.products);
  return (
    <ContextDataCart.Provider value={{ dataCartValue, setDataCartValue }}>
      <Carrinho />
    </ContextDataCart.Provider>
  );
};

const makeSut = (props: any) => {
  return render(<Component {...props} />);
};
describe("Carrinho", () => {
  test("Renderizar listagem de produtos", () => {
    const { container } = makeSut({
      products: dataProduct.map((product) => ({
        ...product,
        imagem: product.imagens,
      })),
    });
    expect(container).toMatchSnapshot("listagem de produtos");
  });
});
