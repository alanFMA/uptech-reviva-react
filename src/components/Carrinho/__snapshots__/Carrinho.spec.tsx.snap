// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carrinho Renderizar listagem de produtos: listagem de produtos 1`] = `
<div>
  <main
    class="principal"
  >
    <h2>
      Minha sacola
    </h2>
    <div
      class="box"
    >
      <div>
        <div
          class="carrinho__grid"
        >
          <div
            class="carrinho__grid__imagem"
          >
            <img
              alt=""
              src="https://static.riachuelo.com.br/RCHLO/13776541005/portrait/8388d930f7a0ad0df4cafcb55355ac1ccc72d06a.jpg"
            />
          </div>
          <div
            class="carrinho__grid__info"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Produto
            </p>
            <p>
              Calça Black Jeans Feminina Skinny Preto
            </p>
          </div>
          <div
            class="carrinho__grid__tamanho"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Tamanho Escolhido
            </p>
            <fieldset
              class="carrinho__grid__tamanho__select"
            >
              <input
                id="p5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="p5"
                >
                  P
                </label>
              </button>
              <input
                id="m5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="m5"
                >
                  M
                </label>
              </button>
              <input
                id="g5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="g5"
                >
                  G
                </label>
              </button>
            </fieldset>
          </div>
          <div
            class="carrinho__valor"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Valor
            </p>
            <p
              class="carrinho__grid__valor"
            >
              55.9
            </p>
          </div>
          <div
            class="carrinho__quantidade"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Quantidade
            </p>
            <input
              id="1"
              max="12"
              min="1"
              name="qtd"
              type="number"
              value=""
            />
          </div>
          <div
            class="subtotal"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Subtotal
            </p>
            <p
              class="carrinho__grid__valor"
            >
              NaN
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="carrinho__grid"
        >
          <div
            class="carrinho__grid__imagem"
          >
            <img
              alt=""
              src="https://static.riachuelo.com.br/RCHLO/14190621002/portrait/57c98408b64b494992da8f5a0661631b07b24e3d.jpg"
            />
          </div>
          <div
            class="carrinho__grid__info"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Produto
            </p>
            <p>
              Camisa Manga Curta Masculina com Bolso
            </p>
          </div>
          <div
            class="carrinho__grid__tamanho"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Tamanho Escolhido
            </p>
            <fieldset
              class="carrinho__grid__tamanho__select"
            >
              <input
                id="p5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="p5"
                >
                  P
                </label>
              </button>
              <input
                id="m5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="m5"
                >
                  M
                </label>
              </button>
              <input
                id="g5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="g5"
                >
                  G
                </label>
              </button>
            </fieldset>
          </div>
          <div
            class="carrinho__valor"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Valor
            </p>
            <p
              class="carrinho__grid__valor"
            >
              75.99
            </p>
          </div>
          <div
            class="carrinho__quantidade"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Quantidade
            </p>
            <input
              id="2"
              max="12"
              min="1"
              name="qtd"
              type="number"
              value=""
            />
          </div>
          <div
            class="subtotal"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Subtotal
            </p>
            <p
              class="carrinho__grid__valor"
            >
              NaN
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="carrinho__grid"
        >
          <div
            class="carrinho__grid__imagem"
          >
            <img
              alt=""
              src="https://static.riachuelo.com.br/RCHLO/14630281001/portrait/286efb86fbe3668e2ec65c06a921a6aac37ff882.jpg"
            />
          </div>
          <div
            class="carrinho__grid__info"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Produto
            </p>
            <p>
              Vestido Curto Feminino Evasê Recorte Floral
            </p>
          </div>
          <div
            class="carrinho__grid__tamanho"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Tamanho Escolhido
            </p>
            <fieldset
              class="carrinho__grid__tamanho__select"
            >
              <input
                id="p5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="p5"
                >
                  P
                </label>
              </button>
              <input
                id="m5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="m5"
                >
                  M
                </label>
              </button>
              <input
                id="g5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="g5"
                >
                  G
                </label>
              </button>
            </fieldset>
          </div>
          <div
            class="carrinho__valor"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Valor
            </p>
            <p
              class="carrinho__grid__valor"
            >
              109.99
            </p>
          </div>
          <div
            class="carrinho__quantidade"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Quantidade
            </p>
            <input
              id="3"
              max="12"
              min="1"
              name="qtd"
              type="number"
              value=""
            />
          </div>
          <div
            class="subtotal"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Subtotal
            </p>
            <p
              class="carrinho__grid__valor"
            >
              NaN
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="carrinho__grid"
        >
          <div
            class="carrinho__grid__imagem"
          >
            <img
              alt=""
              src="https://static.riachuelo.com.br/RCHLO/14327945004/portrait/93dd1b2ac01f6d220c04d73d99633e7f851a83b5.jpg"
            />
          </div>
          <div
            class="carrinho__grid__info"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Produto
            </p>
            <p>
              Camisa Polo Masculina Básica Cinza Turigan
            </p>
          </div>
          <div
            class="carrinho__grid__tamanho"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Tamanho Escolhido
            </p>
            <fieldset
              class="carrinho__grid__tamanho__select"
            >
              <input
                id="p5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="p5"
                >
                  P
                </label>
              </button>
              <input
                id="m5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="m5"
                >
                  M
                </label>
              </button>
              <input
                id="g5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="g5"
                >
                  G
                </label>
              </button>
            </fieldset>
          </div>
          <div
            class="carrinho__valor"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Valor
            </p>
            <p
              class="carrinho__grid__valor"
            >
              84.99
            </p>
          </div>
          <div
            class="carrinho__quantidade"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Quantidade
            </p>
            <input
              id="4"
              max="12"
              min="1"
              name="qtd"
              type="number"
              value=""
            />
          </div>
          <div
            class="subtotal"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Subtotal
            </p>
            <p
              class="carrinho__grid__valor"
            >
              NaN
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="carrinho__grid"
        >
          <div
            class="carrinho__grid__imagem"
          >
            <img
              alt=""
              src="https://static.riachuelo.com.br/RCHLO/14263661004/portrait/f8990b7e94ada9dbee7ddbc4a0e7996f61ce1cf7.jpg"
            />
          </div>
          <div
            class="carrinho__grid__info"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Produto
            </p>
            <p>
              Bermuda Água Masculina Tactel Verão Laranja
            </p>
          </div>
          <div
            class="carrinho__grid__tamanho"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Tamanho Escolhido
            </p>
            <fieldset
              class="carrinho__grid__tamanho__select"
            >
              <input
                id="p5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="p5"
                >
                  P
                </label>
              </button>
              <input
                id="m5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="m5"
                >
                  M
                </label>
              </button>
              <input
                id="g5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="g5"
                >
                  G
                </label>
              </button>
            </fieldset>
          </div>
          <div
            class="carrinho__valor"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Valor
            </p>
            <p
              class="carrinho__grid__valor"
            >
              36.99
            </p>
          </div>
          <div
            class="carrinho__quantidade"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Quantidade
            </p>
            <input
              id="5"
              max="12"
              min="1"
              name="qtd"
              type="number"
              value=""
            />
          </div>
          <div
            class="subtotal"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Subtotal
            </p>
            <p
              class="carrinho__grid__valor"
            >
              NaN
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="carrinho__grid"
        >
          <div
            class="carrinho__grid__imagem"
          >
            <img
              alt=""
              src="https://static.riachuelo.com.br/RCHLO/14196581004/portrait/d6d410bba3e559aad1dac28583863508327d0965.jpg"
            />
          </div>
          <div
            class="carrinho__grid__info"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Produto
            </p>
            <p>
              Bermuda Água Masculina Tactel Verão Azul
            </p>
          </div>
          <div
            class="carrinho__grid__tamanho"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Tamanho Escolhido
            </p>
            <fieldset
              class="carrinho__grid__tamanho__select"
            >
              <input
                id="p5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="p5"
                >
                  P
                </label>
              </button>
              <input
                id="m5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="m5"
                >
                  M
                </label>
              </button>
              <input
                id="g5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="g5"
                >
                  G
                </label>
              </button>
            </fieldset>
          </div>
          <div
            class="carrinho__valor"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Valor
            </p>
            <p
              class="carrinho__grid__valor"
            >
              36.99
            </p>
          </div>
          <div
            class="carrinho__quantidade"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Quantidade
            </p>
            <input
              id="6"
              max="12"
              min="1"
              name="qtd"
              type="number"
              value=""
            />
          </div>
          <div
            class="subtotal"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Subtotal
            </p>
            <p
              class="carrinho__grid__valor"
            >
              NaN
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="carrinho__grid"
        >
          <div
            class="carrinho__grid__imagem"
          >
            <img
              alt=""
              src="https://static.riachuelo.com.br/RCHLO/14428474001/portrait/7ef03efc4422fdf3784543f5598dde8b054dbc05.jpg"
            />
          </div>
          <div
            class="carrinho__grid__info"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Produto
            </p>
            <p>
              Camiseta Masculina Manga Curta Verão Branco
            </p>
          </div>
          <div
            class="carrinho__grid__tamanho"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Tamanho Escolhido
            </p>
            <fieldset
              class="carrinho__grid__tamanho__select"
            >
              <input
                id="p5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="p5"
                >
                  P
                </label>
              </button>
              <input
                id="m5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="m5"
                >
                  M
                </label>
              </button>
              <input
                id="g5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="g5"
                >
                  G
                </label>
              </button>
            </fieldset>
          </div>
          <div
            class="carrinho__valor"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Valor
            </p>
            <p
              class="carrinho__grid__valor"
            >
              42.99
            </p>
          </div>
          <div
            class="carrinho__quantidade"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Quantidade
            </p>
            <input
              id="7"
              max="12"
              min="1"
              name="qtd"
              type="number"
              value=""
            />
          </div>
          <div
            class="subtotal"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Subtotal
            </p>
            <p
              class="carrinho__grid__valor"
            >
              NaN
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="carrinho__grid"
        >
          <div
            class="carrinho__grid__imagem"
          >
            <img
              alt=""
              src="https://static.riachuelo.com.br/RCHLO/14106116003/portrait/180bbff6bbf9b6b492095e3a664c748f81d1e095.jpg"
            />
          </div>
          <div
            class="carrinho__grid__info"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Produto
            </p>
            <p>
              Maiô Tradicional Verão Azul Turquesa
            </p>
          </div>
          <div
            class="carrinho__grid__tamanho"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Tamanho Escolhido
            </p>
            <fieldset
              class="carrinho__grid__tamanho__select"
            >
              <input
                id="p5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="p5"
                >
                  P
                </label>
              </button>
              <input
                id="m5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="m5"
                >
                  M
                </label>
              </button>
              <input
                id="g5"
                name="tamanho5"
                type="radio"
              />
              <button
                class="carrinho__tamanho__botao"
              >
                <label
                  for="g5"
                >
                  G
                </label>
              </button>
            </fieldset>
          </div>
          <div
            class="carrinho__valor"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Valor
            </p>
            <p
              class="carrinho__grid__valor"
            >
              36.99
            </p>
          </div>
          <div
            class="carrinho__quantidade"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Quantidade
            </p>
            <input
              id="8"
              max="12"
              min="1"
              name="qtd"
              type="number"
              value=""
            />
          </div>
          <div
            class="subtotal"
          >
            <p
              class="carrinho__grid__titulo"
            >
              Subtotal
            </p>
            <p
              class="carrinho__grid__valor"
            >
              NaN
            </p>
          </div>
        </div>
      </div>
      <div
        class="total"
      >
        <p
          class="total__titulo"
        >
          Total
        </p>
        <p
          class="calc__total"
        >
          R$ 
          NaN
        </p>
        <p
          class="calc__subtotal"
        >
           x 
          55.9
           = 
          NaN
        </p>
        <p
          class="calc__subtotal"
        >
           x 
          75.99
           = 
          NaN
        </p>
        <p
          class="calc__subtotal"
        >
           x 
          109.99
           = 
          NaN
        </p>
        <p
          class="calc__subtotal"
        >
           x 
          84.99
           = 
          NaN
        </p>
        <p
          class="calc__subtotal"
        >
           x 
          36.99
           = 
          NaN
        </p>
        <p
          class="calc__subtotal"
        >
           x 
          36.99
           = 
          NaN
        </p>
        <p
          class="calc__subtotal"
        >
           x 
          42.99
           = 
          NaN
        </p>
        <p
          class="calc__subtotal"
        >
           x 
          36.99
           = 
          NaN
        </p>
        <button
          class="total__botao"
        >
          Ir para pagamento
        </button>
      </div>
    </div>
  </main>
</div>
`;
